applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_base_element.h

```diff
@@ -0,0 +1,471 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Armin Geiser, https://github.com/armingeiser
+//
+
+#if !defined(ADJOINT_FINITE_DIFFERENCE_BASE_ELEMENT_H_INCLUDED )
+#define  ADJOINT_FINITE_DIFFERENCE_BASE_ELEMENT_H_INCLUDED
+
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/element.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+///@}
+
+///@name Type Definitions
+///@{
+///@}
+
+///@name  Enum's
+///@{
+///@}
+
+///@name  Functions
+///@{
+///@}
+
+///@name Kratos Classes
+///@{
+
+/** \brief AdjointFiniteDifferencingBaseElement
+ *
+ * This element a wrapper for a primal element to calculate element derivatives using
+ * finite differencing  (adjoint semi analytic approach). It is designed to be used in adjoint
+ * sensitivity analysis
+ */
+class AdjointFiniteDifferencingBaseElement : public Element
```

Comment:
since this is the baseclass, can you please add descriptions to the functions that are not in `element.h`?